testing/btest/language/common-mistakes2.zeek

```diff
@@ -0,0 +1,30 @@
+# A companion tonguage/common-mistakes.zeek.  Split off because we skip this
+# test when using ZAM, since it employs a type-checking violation via
+# vector-of-any, which doesn't seem worth going out of our way to support
+# in ZAM (and it isn't dead simple to do so).
+
+# @TEST-REQUIRES: test "${ZEEK_ZAM}" != "1"
+
+# @TEST-EXEC: zeek -b 3.zeek >3.out 2>3.err
+# @TEST-EXEC: btest-diff 3.out
+# @TEST-EXEC: btest-diff 3.err
+
+@TEST-START-FILE 3.zeek
```

Comment:
This doesn't really need to be put into it's own `@TEST-START-FILE` block here.

Reply:
Likewise will be fixed.